lesson32-home-work-43/src/components/ContactsTable.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState} from 'react';
+import Form from './Form';
+import jsonData from '../data/userData.json';
+
+const ContactsTable = () => {
+    const [contactsData, setContactsData] = useState(jsonData);
+    const [showForm, setShowForm] = useState(false);
+
+    const handleDelete = id => {
+        const updatedContacts = contactsData.filter(contact => contact.id !== id);
+        setContactsData(updatedContacts);
+    }
+
+    const tableRows = contactsData.map((info) => {
+        return (
+            <tr key={info.id}>
+                <td>{info.name}</td>
+                <td>{info.surname}</td>
+                <td>{info.phone}</td>
+                <td>
+                    <button className='remove-btn' onClick={() => handleDelete(info.id)}>Delete</button>
+                </td>
+            </tr>
+        )
+    })
+
+    const handleSave = (data) => {
+        data.id = Date.now();
+        const updatedContactsData = [...contactsData];
+        updatedContactsData.push(data);
+        setContactsData(updatedContactsData);
+    }

```

Comment:
Don't mutate object `data` argument, bad practice, if you need, you can copy this object and then change the copy
```
// Just use spread
const copy = { ...data, id: Date.now() }

Or object assign
const copy = Object.assign({}, data, { id: Date.now() })

// This you full code inside handleSave
const handleSave = (data) => {
  setContactsData(state => [...state, {...data, id: Date.now()}]);
}
```